### PR TITLE
Add support for -debug tagged container images.

### PR DESCRIPTION
**Pull request by ericgraf**
Issue: https://github.com/giantswarm/giantswarm/issues/29964
This change supports the case where the image name is `default: gcr.io/kaniko-project/executor:v1.18.0-debug`.

### CODE REVIEW

File: default.json5 [hunk 1]
```diff
@@ -65,6 +65,7 @@
         "repo: (?<depName>.*)\n(.+)VERSION=v?(?<currentValue>.+)\\/.*",
         "repo: (?<depName>.*)\n(\\s)*version: v?(?<currentValue>.*?)\n",
         "repo: (?<depName>.*)\n(\\s)*version:(\\s)*v?(?<currentValue>.*?)(\\s)*\n",
+        "repo: (?<depName>.*)\n(\\s*)default: \"?.+:v?(?<currentValue>.*?)\"?-debug\n",
```
Comment (AverageMarcus) [inline default.json5:68]:
```suggestion
        "repo: (?<depName>.*)\n(\\s*)default: \"?.+:v?(?<currentValue>.*?)(-debug)?\"?\n",
```

File: default.json5 [hunk 2]
```diff
@@ -65,6 +65,7 @@
         "repo: (?<depName>.*)\n(.+)VERSION=v?(?<currentValue>.+)\\/.*",
         "repo: (?<depName>.*)\n(\\s)*version: v?(?<currentValue>.*?)\n",
         "repo: (?<depName>.*)\n(\\s)*version:(\\s)*v?(?<currentValue>.*?)(\\s)*\n",
+        "repo: (?<depName>.*)\n(\\s*)default: \"?.+:v?(?<currentValue>.*?)(-debug)?\"?\n",
         "repo: (?<depName>.*)\n(\\s*)default: \"?.+:v?(?<currentValue>.*?)\"?\n",
```
Comment (AverageMarcus) [inline default.json5:69]: You actually dont need this line now.

Comment (ericgraf) [inline default.json5:69]:
```suggestion
        "",
```

Comment (ericgraf) [reply default.json5]: removed

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**